workers/social/lib/social/models/computeproviders/stacktemplate.coffee

```diff
@@ -322,7 +339,76 @@ module.exports = class JStackTemplate extends Module
         callback err, this
 
 
-  generateStack: permit
+  shareWithTeam: permit
+
+    advanced: [
+      { permission: 'update own stack template', validateWith: Validators.own }
+      { permission: 'update stack template' }
+    ]
+
+    success: revive
+
+      shouldReviveClient   : yes
+      shouldReviveProvider : no
+      shouldFetchGroupLimit : yes
+
+    , (client, callback) ->
+
+      { group }    = client.r
+      { delegate } = client.connection
+      { stackTemplates } = group
+
+      stackTemplates = makeArray stackTemplates
+      id = @getId().toHexString()
+
+      if stackTemplates.length
+        unless id in stackTemplates
+          stackTemplates.push id
+      else stackTemplates = [id]
+
+      console.log 'stackTemplates', stackTemplates
+      @setAccess client, 'group', (err) =>
+        group.modify client, { stackTemplates }, (err) =>
+          return callback err  if err
+
+          return callback err, this
```

Comment:
we don't need this method at all, you can simply use `JStackTemplate::setAccess 'group'` from client side. same goes for `makePrivate` method as well.